HIP/HIP-668.md

```diff
@@ -0,0 +1,2349 @@
+  stakedNode: NetworkNode
+
+  """
+  The staking period during which either the staking settings for this account changed (such as starting staking or
+  changing stakedNode) or the most recent reward was earned, whichever is later. If this account is not currently
+  staked to a node, then the value is null.
+  """
+  stakePeriodStart: Timestamp
+
+  "A consensus timestamp range with an inclusive from timestamp and an exclusive to timestamp."
+  timestamp: TimestampRange!
+
+  "The type of Hedera entity."
+  type: EntityType!
+}
+
+"The encoding used to convert the raw bytes to String."
+enum BinaryEncoding {
+  "Base 64 encoding."
+  BASE64
+
+  "Base 16 hexadecimal encoding with a '0x' prefix and using lowercase letters and numbers."
+  HEX
+}
+
+"""
+The base type for an entity on the Hedera network. Each entity is assigned an incrementing identifier that is shared
+across all entity types and associated with a particular shard and realm.
+"""
+interface Entity implements Node {
+  "The account charged the auto-renewal fee."
+  autoRenewAccount: Account
+
+  "The amount of time to elapse before auto-renew occurs."
+  autoRenewPeriod: Duration
+
+  "The consensus timestamp at which the entity was created."
+  createdTimestamp: Timestamp
+
+  "Whether the entity still exists on the ledger."
+  deleted: Boolean
+
+  "The unique identifier associated with the entity."
+  entityId: EntityId!
+
+  """
+  The time at which this entity will expire and be auto-renewed, if possible. If this was not explicitly set by the
+  user it will be calculated as the createdTimestamp plus the autoRenewPeriod.
+  """
+  expirationTimestamp: Timestamp
+
+  "An opaque, globally unique identifier specific to GraphQL."
+  id: ID!
+
+  """
+  The admin key associated with this entity whose signing requirements must be met in order to modify the entity on
+  the network.
+  """
+  key: Object
+
+  "The memo associated with the entity."
+  memo: String!
+
+  "A consensus timestamp range with an inclusive from timestamp and an exclusive to timestamp."
+  timestamp: TimestampRange!
+
+  "The type of Hedera entity."
+  type: EntityType!
+}
+
+"An incrementing identifier that is shared across all entity types and associated with a particular shard and realm."
+type EntityId {
+  "The unique incrementing identifier associated with this entity."
+  num: Long!
+
+  "The realm number. Currently always zero."
+  realm: Long!
+
+  "The shard number to allow for horizontal scaling of networks. Currently always zero."
+  shard: Long!
+}
+
+"The unique identifier to find an entity on the Hedera network."
+input EntityIdInput {
+  "The unique incrementing identifier associated with this entity."
+  num: Long! @Min(value: 0)
+
+  "The realm number. Defaults to zero."
+  realm: Long! = 0 @Min(value: 0)
+
+  "The shard number to allow for horizontal scaling of networks. Defaults to zero."
+  shard: Long! = 0 @Min(value: 0)
+}
```

Comment:
Do people prefer the EntityId as a complex type or due to its wide use in the API prefer it as a `0.0.x` scalar string like the REST API? It can get quite verbose to query for all these entity ids as `accountId {shard, realm, num}`